modules/telestion-api/src/main/java/org/telestion/api/message/JsonMessage.java

```diff
@@ -117,4 +99,28 @@ default String className() {
 	default JsonObject json() {
 		return JsonObject.mapFrom(this);
 	}
+
+	/**
+	 * Reflection part of {@link #on(Class, Message, Handler)} and {@link #on(Class, Object, Handler)}.
+	 *
+	 * @param clazz			Class to cast to
+	 * @param jsonObject	Object to be casted
+	 * @param handler		Handler for the message
+	 * @param <T>			Type of the message
+	 * @return if parsing successful
+	 */
+	private static <T extends JsonMessage> boolean doReflection(Class<T> clazz, JsonObject jsonObject,
```

Comment:
The name is not very precise. While it is generally a good idea to pull out duplicated code you don't necessarily need to introduce a new method. Here it would be better to chain the method call in the second variant with a call like `return on(msg.body(),...)`. You don't need this new method.